servant-client-core/src/Servant/Client/Core/Internal/ClientF.hs

```diff
@@ -0,0 +1,10 @@
+{-# LANGUAGE DeriveFunctor #-}
+module Servant.Client.Core.Internal.ClientF where
+
+import Servant.Client.Core.Internal.Request
+
+data ClientF a
+    = RunRequest Request (Response -> a)
+    | StreamingRequest Request (StreamingResponse -> a)
+    | Throw ServantError
```

Comment:
Maybe adding a `Catch` constructor would help fix the problem with `catchServantError` that you mention below?

Reply:
`catch` isn't an algebraic effect. So it won't really work out well.